tests/testsuite/registry_overlay.rs

```diff
@@ -0,0 +1,161 @@
+//! Tests for local-registry sources.
+
+use cargo_test_support::registry::{Package, RegistryBuilder, TestRegistry};
+use cargo_test_support::{project, Project};
+
+fn setup() -> (TestRegistry, String) {
+    let alt = RegistryBuilder::new().alternative().build();
+    (
+        RegistryBuilder::new().http_index().build(),
+        alt.index_url()
+            .to_file_path()
+            .unwrap()
+            .into_os_string()
+            .into_string()
+            .unwrap(),
+    )
+}
+
+fn simple_project() -> Project {
+    project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                bar = ">= 0.0.0"
+                baz = ">= 0.1.0"
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build()
+}
+
+#[cargo_test]
+fn overlay_hit() {
+    let (reg, alt_path) = setup();
+    let p = simple_project();
+
+    Package::new("bar", "0.0.1").publish();
+    // baz is only in the local registry, but it gets found
+    Package::new("baz", "0.1.1")
+        .alternative(true)
+        .local(true)
+        .publish();
+
+    p.cargo("check")
+        .overlay_registry(&reg.index_url(), &alt_path)
+        .run();
+}
+
+#[cargo_test]
+fn version_fallback() {
+    let (reg, alt_path) = setup();
+    let p = simple_project();
+
+    Package::new("bar", "0.0.1").publish();
```

Comment:
What is the role of `bar` in these tests?   We should either simplify the test cases or name the packages to convey the role they play in the tests

Reply:
I think it's not needed; I'll remove it.